hardkernel/odroid-hc4/default.nix

```diff
@@ -1,22 +1,28 @@
-{ lib, ... }:
+{ config, lib, ... }:
 
 {
   # Based on the config from https://www.armbian.com/odroid-hc4/
   hardware.fancontrol = {
     enable = lib.mkDefault true;
-    config = lib.mkDefault ''
+    config = let
+      # According to https://www.armbian.com/odroid-hc4/ the FCFANS line should be removed on kernel 5.15.
+      kernelVersion = config.boot.kernelPackages.kernel.version;
+      needFcFans = lib.versions.majorMinor kernelVersion != "5.15";
+    in lib.mkDefault (''
       INTERVAL=10
       DEVPATH=hwmon0=devices/virtual/thermal/thermal_zone0 hwmon2=devices/platform/pwm-fan
       DEVNAME=hwmon0=cpu_thermal hwmon2=pwmfan
       FCTEMPS=hwmon2/pwm1=hwmon0/temp1_input
+    '' + lib.optionalString needFcFans ''
       FCFANS= hwmon2/pwm1=hwmon2/fan1_input
+    '' + ''
```

Comment:
Needed to change this because otherwise, the lines would actually get squished together when `needFcFans`.